oeps/best-practices/oep-0068-bp-openfeature-toggles.rst

```diff
@@ -0,0 +1,216 @@
+
+Reference Implementation
+************************
+
+An example implementation of a Python provider that retrieves toggle values from Django settings could look like this:
+
+.. code-block:: python
+
+    from typing import Any, Dict
+    from django.conf import settings
+    from openfeature.provider.provider import AbstractProvider
+    from openfeature.flag_evaluation import FlagResolutionDetails, EvaluationContext
+
+    class DjangoSettingsProvider(AbstractProvider):
+        def __init__(self):
+            super().__init__()
+            self.name = "Django Settings Provider"
+
+        def get_metadata(self):
+            return {"name": self.name}
+
+        def get_bool_value(
+            self,
+            flag_key: str,
+            default_value: bool,
+            evaluation_context: EvaluationContext = None,
+        ) -> FlagResolutionDetails[bool]:
+            value = self._get_flag_value(flag_key, default_value)
+            return FlagResolutionDetails(value=bool(value))
+
+        def get_string_value(
+            self,
+            flag_key: str,
+            default_value: str,
+            evaluation_context: EvaluationContext = None,
+        ) -> FlagResolutionDetails[str]:
+            value = self._get_flag_value(flag_key, default_value)
+            return FlagResolutionDetails(value=str(value))
+
+        def get_integer_value(
+            self,
+            flag_key: str,
+            default_value: int,
+            evaluation_context: EvaluationContext = None,
+        ) -> FlagResolutionDetails[int]:
+            value = self._get_flag_value(flag_key, default_value)
+            return FlagResolutionDetails(value=int(value))
+
+        def get_float_value(
+            self,
+            flag_key: str,
+            default_value: float,
+            evaluation_context: EvaluationContext = None,
+        ) -> FlagResolutionDetails[float]:
+            value = self._get_flag_value(flag_key, default_value)
+            return FlagResolutionDetails(value=float(value))
+
+        def get_object_value(
+            self,
+            flag_key: str,
+            default_value: Dict,
+            evaluation_context: EvaluationContext = None,
+        ) -> FlagResolutionDetails[Dict]:
+            value = self._get_flag_value(flag_key, default_value)
+            return FlagResolutionDetails(value=dict(value))
+
+        def _get_flag_value(self, flag_key: str, default_value: Any) -> Any:
+            return getattr(settings, flag_key, default_value)
+
+
+Usage of the above provider could look like:
+
+.. code-block:: python
+
+    import openfeature.api
+    from django.conf import settings
+
+    # In your Django settings.py
+    FEATURE_FLAG_EXAMPLE = True
+
+    # In your application code
+    client = openfeature.api.get_client("edx")
+    openfeature.api.set_provider(DjangoSettingsProvider())
+
+    # Evaluate a flag
+    is_enabled = client.get_boolean_value("FEATURE_FLAG_EXAMPLE", False)
```

Comment:
This seems to be against best practice - the default value of `FEATURE_FLAG_EXAMPLE` as well as the literal string value of the flag name should be defined in one place, and not potentially defined separately in every place where the value is consumed. I know this example isn't necessarily saying that, but it's ambiguous and we want to discourage that.

> The current approach of annotating toggles with explicit metadata will be maintained.

If that's the case (which is great), I think the example code should reflect that and include the toggle metadata.